skbio/stats/composition.py

```diff
@@ -489,13 +527,36 @@ def clr(mat):
     array([-0.79451346,  0.30409883,  0.5917809 , -0.10136628])
 
     """
-    mat = closure(mat)
-    lmat = np.log(mat)
-    gm = lmat.mean(axis=-1, keepdims=True)
-    return (lmat - gm).squeeze()
+    axis = -1
+    # xp is the namespace wrapper for different array libraries
+    # NOTE: the following (try:) may reduce the efficiency but to
+    # keep backward compatibility when the input is a list
+    try:
+        xp = aac.array_namespace(mat)
+    except Exception as e:
+        mat = np.asarray(mat)
+        xp = aac.array_namespace(mat)
 
 
-def clr_inv(mat):
+    if validate:
+        # assert from closure() while it is removed in latest version
+        if xp.any(mat < 0):
+            raise ValueError("Cannot have negative proportions")
```

Comment:
Same here, isn't his redundant since this check is contained inside of `_composition_check`?